### PR TITLE
CDPT-622 Family Mediators API Ruby upgrade.

### PR DESCRIPTION
**Pull request by njpreddy**
Upgraded the Ruby version from 2.7.2 to 3.1.4.

### CODE REVIEW

File: .rubocop.yml [hunk 1]
```diff
@@ -1,3 +1,7 @@
+require:
+  - rubocop-performance
+  - rubocop-rails
+
 AllCops:
   TargetRubyVersion: 2.5
```
Comment (vertism) [inline .rubocop.yml:6]: This should probably be updated

File: Gemfile.lock [hunk 2]
```diff
@@ -1,24 +1,24 @@
 GEM
   remote: https://rubygems.org/
   specs:
-    activemodel (6.1.7.1)
-      activesupport (= 6.1.7.1)
-    activerecord (6.1.7.1)
-      activemodel (= 6.1.7.1)
-      activesupport (= 6.1.7.1)
-    activesupport (6.1.7.1)
+    activemodel (6.1.7.3)
```
Comment (vertism) [inline Gemfile.lock:4]: Are all the gem updates necessary for Ruby 3.1 to work? 
Updating dry-configurable, sinatra and grape all at the same time feels risky.

File: .rubocop.yml (hunk 1 above)
Comment (vertism) [reply .rubocop.yml:6]:
```suggestion
  TargetRubyVersion: 3.1
```

Comment (njpreddy) [reply .rubocop.yml]: Targer version updated

File: Gemfile.lock (hunk 2 above)
Comment (njpreddy) [reply Gemfile.lock:4]: Grape definitely needed updating in order for the rspec tests to pass.

Comment (vertism) [reply Gemfile.lock:4]: If you replace Gemfile.lock with the one from the main branch and then re-run `bundle install` that should leave you with just the necessary changes

**Review by vertism — APPROVED** [review]
LGTM

**Pull request merged**